src/components/MerchantsPage/gam/CampaignListItem.tsx

```diff
@@ -5,47 +5,84 @@ import {
+import { useEffect, useState } from 'react';
 
 interface Props {
-  campaign: String;
+  campaign: Campaign;
 }
 
-const CampaignListItem = (campaign: Props) => {
+const CampaignListItem = (props: Props) => {
+  const [distributor, setDistributor] = useState<any | null>();
+  const [merchant, setMerchant] = useState<any | null>();
+  const campaign = props.campaign;
+
+  const fetchData = async () => {
+    const distributorData = await getDistributor(campaign.distributor_id);
+    const merchantData = await getSeller(campaign.seller_id);
+    setDistributor(distributorData.data);
+    setMerchant(merchantData.data);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const mealsRaised = Math.floor(campaign.amount_raised / campaign.price_per_meal);
+  const targetMeals = Math.floor(campaign.target_amount / campaign.price_per_meal);
+  const campaignImageUrls = campaign.gallery_image_urls;
+
   return (
     <Container>
       <ColumnContainer>
-        <img src={campaignDefaultImage} alt="campaign_image" />
+        {campaignImageUrls && campaignImageUrls.length && (
+          <img src={campaignImageUrls[0] ?? campaignDefaultImage} alt="campaign_image" />
+        )}
       </ColumnContainer>
       <ColumnContainer>
-        <Location>Sunset Park, Brooklyn</Location>
-        <Name>Melonpanna Tea & Shot x APEX for the Youth</Name>
+        {merchant && merchant.locations && merchant.locations.length && (
+          <Location>{merchant.locations[0].city}</Location>
+        )}
+        {distributor && merchant && (
+          <Name>{merchant.name} x {distributor.name}</Name>
+        )}
         <Description>
-          Partnering with APEX for the Youth, we hope to raise 200 meals for
-          underserved Asian and immigrant youth from low-income families.
+          {campaign.description}
+          <br></br>
+          {distributor &&
+            (<a href={distributor.website_url}>{distributor.name}</a>)
+          }
         </Description>
-        {/* Testing values */}
         <CampaignProgressBar
-          isActive={true}
-          numContributions={73}
-          targetAmount={100}
+          isActive={campaign.active}
+          numContributions={mealsRaised}
+          targetAmount={targetMeals}
           progressBarColor={'#CF6E8A'}
-          lastContributionTime={new Date('07/21/2020 20:05:00')}
-          endDate={new Date('07/23/2020')}
+          lastContributionTime={new Date(campaign.last_contribution)}
+          endDate={new Date(campaign.end_date)}
         />
       </ColumnContainer>
       <ColumnContainer>
         <ImagesContainer>
-          <img src={apexLogo} alt="merchant_logo" />
-          <img src={melonpannaLogo} alt="distributor_logo" />
+          {distributor &&
+            (<a href={distributor.website_url}>
+              <img src={distributor.image_url} alt="distributor_logo" />
+            </a>
+            )
+          }
+          <img src={melonpannaLogo} alt="merchant_logo" />
```

Comment:
Need to expose a "logo_image_url" field on the Seller model, the current hero_image_url and owner_image_url can't be used for this